### ISSUE TITLE
Bug(ish): Kernel buff/cache usage prevents downscaling

### ISSUE THREAD

**Issue by sharnoff** [neondatabase/autoscaling#531]
This is also _kind of_ a feature request.

Basically, buff/cache memory usage is not counted separately from other usage, which means that VMs with non-trivial read loads will tend to balloon in memory usage.

There's two parts to resolving this:

1. The vm-monitor needs to allow downscaling when that may require evicting some memory that's currently being used by the kernel for buff/cache
    - Note: We should be careful here to appropriately factor in "dirty" cache usage, because otherwise downscaling might block on writeback
2. The autoscaler-agent needs to take buff/cache mem usage _somewhat_ into account, so that we can maintain _some_ cache available to be used, but not too too much.

```[tasklist]
# Tasks
- [ ] Change vm-monitor to ignore buff/cache memory usage when considering downscale
- [ ] Change autoscaler-agent to _partially_ consider buff/cache usage when deciding target memory usage
```

Probably easiest to implement this after #371.

**Comment by sharnoff** [comment]
This should actually be _mostly_ fixed by neondatabase/neon#5524, now that #371 is merged so the agent retries downscale requests with progressively higher CU.

That fix will still generate a lot of extra traffic, so we'll still want the changes in the agent, unfortunately.

**Comment by sharnoff** [comment]
I think this is actually already fixed. There's still imperfections (agent often disagrees with vm-monitor), but they're mostly intentional for now, pending a larger redesign.
